bindings/rust/s2n-tls-tokio/tests/shutdown.rs

```diff
@@ -81,3 +83,87 @@ async fn shutdown_after_split() -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+#[tokio::test(start_paused = true)]
+async fn shutdown_with_blinding() -> Result<(), Box<dyn std::error::Error>> {
+    let client = TlsConnector::new(common::client_config()?.build()?);
+    let server = TlsAcceptor::new(common::server_config()?.build()?);
+
+    let (server_stream, client_stream) = common::get_streams().await?;
+    let server_stream = common::TestStream::new(server_stream);
+    let overrides = server_stream.overrides();
+    let (mut client, mut server) =
+        common::run_negotiate(&client, client_stream, &server, server_stream).await?;
+
+    // Trigger a blinded error.
+    overrides.next_read(Some(Box::new(|_, _, buf| {
+        // Parsing the header is one of the blinded operations
+        // in s2n_recv, so provide a malformed header.
+        let zeroed_header = [23, 0, 0, 0, 0];
+        buf.put_slice(&zeroed_header);
+        Ok(()).into()
+    })));
+    let mut received = [0; 1];
+    let result = server.read_exact(&mut received).await;
+    assert!(result.is_err());
+
+    // Shutdown MUST NOT complete faster than minimal blinding time.
+    let (timeout, _) = join!(
+        time::timeout(common::MIN_BLINDING_SECS, server.shutdown()),
+        time::timeout(common::MIN_BLINDING_SECS, read_until_shutdown(&mut client)),
+    );
+    assert!(timeout.is_err());
+
+    // Shutdown MUST eventually gracefully complete after blinding
+    let (timeout, _) = join!(
```

Comment:
:( Isn't this the same thing as line 111? Why are we expecting a different result?

Reply:
Cleaned up the comment. We're not checking for success here, just not a timeout.